### PR TITLE
fix: fixed fetch bug on first render of id page

### PR DESCRIPTION
**Pull request by oskarikerppo**

### CODE REVIEW

**Comment by oskarikerppo** [comment]
Network traffic upon refreshing http://localhost:3000/projects/1 before this fix
Screenshot from 2022-01-20 11-21-04
:

**Comment by oskarikerppo** [comment]
After fix: 
Screenshot from 2022-01-20 11-21-41

**Comment by oskarikerppo** [comment]
To clarify, upon refreshing page the `router.query` was undefined. This caused problems in the `useData` hook.

File: lib/utils/fetch.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { Auth } from 'aws-amplify';
 
 // https://eckertalex.dev/blog/typescript-fetch-wrapper
-async function http<T>(path: string, config?: RequestInit): Promise<T> {
+async function http<T>(path?: string, config?: RequestInit): Promise<T> {
```
Comment (LeadingMoominExpert) [inline lib/utils/fetch.ts:4]: I'm not a huge fan of the path being nullable. Having a nullable function parameter here just adds complexity to the function which doesn't need it.

What's the case that requires converting path from undefined to empty string?

Comment (LeadingMoominExpert) [reply lib/utils/fetch.ts:4]: Based on you screenshots there's no api queries ran to empty path?

Comment (oskarikerppo) [reply lib/utils/fetch.ts:4]: > Based on you screenshots there's no api queries ran to empty path?

The first screenshot is from the main branch. Second screenshot there's not api queries with empty path as supposedly fetch catches the error with empty path

**Pull request closed without merging**